include/zephyr/net/net_filter.h

```diff
@@ -0,0 +1,126 @@
+/*
+ * Copyright (c) 2023 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/** @file
+ * @brief Public API for network filter
+ *
+ * The simple network filter allows to network-related operations
+ * which can be implemented in the form of hooks - custom callbacks.
+ */
+
+#ifndef ZEPHYR_INCLUDE_NET_FILTER_H_
+#define ZEPHYR_INCLUDE_NET_FILTER_H_
+
+#include <zephyr/net/net_core.h>
+#include <zephyr/net/net_ip.h>
+#include <zephyr/net/net_pkt.h>
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/**
+ * @brief Network Filter API
+ * @defgroup net_filter Network Filter API
+ * @ingroup networking
+ * @{
+ */
+
+/** @brief hook callback type. */
+typedef enum net_verdict (nf_hook_fn_t)(struct net_pkt *pkt);
+
+/** @brief Hook IDs for IP. */
+typedef enum {
+	NF_IP_PRE_ROUTING = 0,
+	NF_IP_LOCAL_IN,
+	NF_IP_FORWARD,
+	NF_IP_LOCAL_OUT,
+	NF_IP_POST_ROUTING,
+	NF_IP_NUMHOOKS
+} nf_ip_hooks;
```

Comment:
Maybe `ip_hook_type` would be a bit more descriptive?

Reply:
This is a hook type categories for IP hooks, I'm not sure if `ip_hook_type` it's a good name for this enum.
If it is a case I can change it to list of #define-s